src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +194,1374 @@ static DamageTable D_80BBF858 = {
+        this->actionFunc = EnSGoro_ActionShopGoronTalking;
+    }
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000, (s16)0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+}
+
+// func_80BBECBC -- Action func.
+void EnSGoro_ActionShopGoronTalking(EnSGoro* this, PlayState* playState) {
+
+    u8 msgState = Message_GetState(&playState->msgCtx);
+
+    if (FLAG_IS_SET(this->actionFlags, ENSGORO_ACTIONFLAG_HANDTAP)) {
+        EnSGoro_UpdateToHandtapAnimation(this);
+    } else if (FLAG_IS_SET(this->actionFlags, ENSGORO_ACTIONFLAG_TIRED)) {
+        EnSGoro_UpdateSleepyAnimation(this);
+    } else {
+        EnSGoro_UpdateToIdleAnimation(this);
+    }
+
+    if (msgState == 6) {
+        if (Message_ShouldAdvance(playState)) {
+            if (FLAG_IS_SET(this->actionFlags, ENSGORO_ACTIONFLAG_LASTMESSAGE)) {
+                SET_FLAG_OFF(this->actionFlags, ENSGORO_ACTIONFLAG_LASTMESSAGE);
+                SET_FLAG_OFF(this->actionFlags, ENSGORO_ACTIONFLAG_ENGAGED);
+                this->textID = EnSGoro_Msg_None;
+                this->actionFunc = EnSGoro_ActionShopGoron;
+                return;
+            }
+            this->textID = EnSGoro_BombshopGoron_NextTextID(this, playState);
+            Message_StartTextbox(playState, this->textID, &this->actor);
+        }
+    } else if ((msgState == 4) && (Message_ShouldAdvance(playState))) {
+        switch (playState->msgCtx.choiceIndex) {
+            case 0:
+                SET_FLAG_ON(this->bombbuyFlags, ENSGORO_BOMBBUYFLAG_YESBUY);
+                break;
+            case 1:
+                func_8019F230();
+                SET_FLAG_OFF(this->bombbuyFlags, ENSGORO_BOMBBUYFLAG_YESBUY);
+                break;
+        }
+
+        this->textID = EnSGoro_BombshopGoron_NextTextID(this, playState);
+        if ((this->textID == EnSGoro_Msg_Shop_GoronPK_YesBuy) ||
+            (this->textID == EnSGoro_Msg_Shop_GoronPK_YesBuy_LastNight)) {
+            playState->msgCtx.msgMode = 0x43;
+            playState->msgCtx.unk12023 = 4;
+            Actor_PickUp(&this->actor, playState, 0x34, 300.0f,
+                         300.0f); // 0x34 is probably powder keg, but not currently listed in enum...
+            this->actionFunc = EnSGoro_ActionShopTakePayment;
+        } else {
+            Message_StartTextbox(playState, this->textID, &this->actor);
+        }
+    }
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000, (s16)0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+}
+
+// func_80BBEEB4 -- Action func.
+void EnSGoro_ActionShopTakePayment(EnSGoro* this, PlayState* playState) {
+    if (Actor_HasParent(&this->actor, playState)) {
+        this->actor.parent = NULL;
+        func_801159EC(this->powderKegPrice * -1);
+        this->actionFunc = EnSGoro_ActionShopFinishTransaction;
+        return;
+    }
+    Actor_PickUp(&this->actor, playState, 0x34, 300.0f, 300.0f);
```

Comment:
```suggestion
    Actor_PickUp(&this->actor, playState, GI_POWDER_KEG, 300.0f, 300.0f);
```
The third argument (arg2) is a GetItemId which should use the `GetItemID` enum in z64items.h. Since it doesn't exist yet, we can add a new entry. (Note it looks like it is going to be added in #900 )